src/b2bRoutes.ts

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright © 2019 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+import { lazy } from 'react';
+
+export default function b2bRoutes() {
+  /** All B2B Routes */
+  const Accounts = import(/* webpackChunkName: "Accounts" */ './containers/b2b/Accounts');
+  const AccountMain = import(/* webpackChunkName: "AccountMain" */ './containers/b2b/AccountMain');
+  const RequisitionList = import(/* webpackChunkName: "RequisitionList" */ './containers/b2b/RequisitionList');
+  const RequisitionPageMain = import(/* webpackChunkName: "RequisitionPageMain" */ './containers/b2b/RequisitionPageMain');
+
+  /** All B2C routes */
+  /** Place the most likely first-visit-pages to be fetched first! */
+  const HomePage = import(/* webpackChunkName: "HomePage" */ './containers/HomePage');
+  const CartPage = import(/* webpackChunkName: "CartPage" */ './containers/CartPage');
+  const ProductDetailPage = import(/* webpackChunkName: "ProductDetailPage" */ './containers/ProductDetailPage');
+  const ProfilePage = import(/* webpackChunkName: "ProfilePage" */ './containers/ProfilePage');
+  const CategoryPage = import(/* webpackChunkName: "CategoryPage" */ './containers/CategoryPage');
```

Comment:
We shouldn't break our base components like CategoryPage into separate chunks. This will trigger many fetches in order to just show categories/products/cart and our own components aren't so big anyways. I think the better strategy would be to separate chunks into:
- Chunk 1: Our "base" components ie. everything you need to do the most basic commerce.
- Chunk 2: Our "additional" components, like b2b and everything you don't absolutely have to have for the basic commerce.
- All large imports that are not required for Chunk 1 should be one per chunk (think aframe and other large imports).
- Webpack will automatically bundle all remaining imports into the "vendor" bundle.

This way we optimize for great UX when all you need is to visit site/make a quick purchase etc. This is critical as most people do just that especially on mobile where network might be slower. If you need b2b you are probably on desktop anyway and you have different expectations. Same goes for aframe and other integrations, if you click on a 3d model you kind of expect it will take tame to load.

Let's discuss this further on our standup tomorrow.

Reply:
This approach may slow down that initial first pixel that the user sees because the first chunk that is loaded is larger.  Although I do agree that the number of asynchronous requests may have become too high here.  I'll revisit this and try and find a good balance between numberof asynchonous requests vs bundle size.  Hopefully I find a balance with compelling numbers.

My initial thoughts are that we bundle all routes and even components within those routes that may be likely to be on that first load.  For example more likely routes may include the homepage and PDP (someone sends a product item to show someone), less likely might be a profile page or any routes relating to the checkout flow as this indicates the user has already made a visit to the site and thus chunks have been cached in the users browser.